cert/selfsigned.go

```diff
@@ -0,0 +1,172 @@
+package cert
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"net"
+	"os"
+	"time"
+)
+
+const (
+	// AutogenCertValidity is the default validity of a self-signed
+	// certificate. The value corresponds to 14 months
+	// (14 months * 30 days * 24 hours).
+	AutogenCertValidity = 14 * 30 * 24 * time.Hour
+)
+
+var (
+	// End of ASN.1 time.
+	endOfTime = time.Date(2049, 12, 31, 23, 59, 59, 0, time.UTC)
+
+	// Max serial number.
+	serialNumberLimit = new(big.Int).Lsh(big.NewInt(1), 128)
+)
+
+// GenCertPair generates a key/cert pair to the paths provided. The
+// auto-generated certificates should *not* be used in production for public
+// access as they're self-signed and don't necessarily contain all of the
+// desired hostnames for the service. For production/public use, consider a
+// real PKI.
+//
+// This function is adapted from https://github.com/btcsuite/btcd and
+// https://github.com/btcsuite/btcutil
+func GenCertPair(org, certFile, keyFile string, tlsExtraIPs,
```

Comment:
This is a clean move that only exports the funcs? Diff doesn't show diffs. An alternative way that we sometimes use is to do a commit first that is only an unchanged 1:1 copy. Then in the next commit, make the required changes (export, remove old code, reference `cert`)

Reply:
Changed commit structure to address this.

Reply:
Yes, much more confident about the changes with this new structure